timetracker/entries/models.py

```diff
@@ -1,25 +1,33 @@
 from django.db import models
-
 from django.utils import timezone
 
+class Client(models.Model):
+    name = models.CharField(max_length=200)
+
+    def __str__(self):
+        return self.name
 
 class Project(models.Model):
-    client = models.CharField(max_length=200)
+    client = models.ForeignKey('Client')
     name = models.CharField(max_length=200)
 
     def __str__(self):
         return '<{}> {}'.format(self.client, self.name)
 
-
-# Create your models here.
-class Entry(models.Model):
+class WorkPeriod(models.Model):
```

Comment:
Whilst I can see where your reasoning has gone here, separating WorkPeriod here doesn't really fit.

The reason is that the start/end time are specific to the Entry, and if two Entries share a start or end time it's purely coincidental, not meaningful.